### DISCUSSION TITLE
Renovate does not help if the GitLab User does not have enough permissions for automerge

### DISCUSSION THREAD

**Discussion by Morl99** [renovatebot/renovate#22169] [Suggest an Idea]
**What Renovate type, platform and version are you using?**
Self hosted Renovate on a self hosted GitLab CE 13.8.3
Renovate is run via a docker container in a scheduled pipeline, current renovate version (changing daily ;)): 24.83.0

**Describe the bug**
This one took me some time to hunt down. After creating a few new projects and configuring them in our renovate pipeline, we noticed, that automerge did not work for them. This was odd, because automerge worked for alle the other projects (where it is enabled). While the automerge config was shown as enabled in the PR, we had to manually merge those requests.

We finally figured out, that the technical user in those repos only had Developer permissions, and thus, was not allowed to merge. While this of course is a configuration issue on our end, some kind of pointer to the issue would have been nice. Ideally, Renovate would create an issue about this, but at least an error in the logs is needed. If automerge is enabled, but the user does not have the permission to merge, this is a clear configuration error, that renovate should help to detect.

**Relevant debug logs**
The logs did not help, all they showed was:

```
DEBUG: Checking #22 for automerge (repository=XXX, branch=renovate/express-openapi-validator-4.x)
       "automerge": true,
       "automergeType": "pr",
       "automergeComment": "automergeComment"
DEBUG: PR is configured for automerge (repository=XXX, branch=renovate/express-openapi-validator-4.x)
DEBUG: PR is not ready for merge (repository=XXX, branch=renovate/express-openapi-validator-4.x)
       "canMergeReason": undefined
```

**Have you created a minimal reproduction repository?**
Not sure what I would do here, because this is related to the repository configuration, and not the dependencies or the renovate config. So I assume, that a mrr is not strictly needed. Please let me know if you disagree.

- [X] This is a really small bug, it does not need a reproduction (think small typo)

**Comment by HonkingGoose** [comment]
I'd say this is more of a feature request, so labeling it as such. 😄

**Comment by rarkins** [comment]
Needs reproduction and debugging by someone with this setup so that we can know which logic flow is covered within lib/platform/gitlab/*. It would seem that `pr.canMerge` is not `true` but we don't know if that's something GitLab is returning to us gracefully or via an error.

**Comment by rarkins** [comment]
This is not really "needs reproduction" but instead someone with this setup needs to do the work if they want this implemented.
